frontend/src/screens/HomeScreen.tsx

```diff
@@ -1,32 +1,21 @@
-import React, { useEffect } from 'react'
-import { useDispatch, useSelector } from 'react-redux'
 import { Col, Row } from 'react-bootstrap'
 import { Product } from '../components/Product'
-import { Dispatch } from 'redux'
 import { Message } from '../components/Message'
 import { Loader } from '../components/Loader'
-import { listProducts } from '../actions/productActions'
+import { useQuery } from '@apollo/client'
+import { GET_ALL_PRODUCTS } from '../graphql/queries/product/product-query'
 
 export const HomeScreen = () => {
-  const dispatch: Dispatch<any> = useDispatch()
+  const { loading, error, data: products } = useQuery(GET_ALL_PRODUCTS)
 
-  type ProductListState = {
-    productList: {
-      loading: boolean
-      error: string
-      products: any[]
-    }
+  type Product = {
```

Comment:
It’s usually a good idea to have a folder from which you export all types

Reply:
done